src/replica/python/schema_migration.py

```diff
@@ -53,19 +51,14 @@ class MasterReplicationMigrationManager(SchemaMigMgr):
 
     def __init__(
         self,
-        name: str,
         connection: str,
         scripts_dir: str,
         set_initial_configuration: Optional[Callable[[], None]],
         repl_connection: Optional[str],
     ):
         self.repl_connection = repl_connection
-        self.set_initial_configuartion = set_initial_configuration
+        self.set_initial_configuration = set_initial_configuration
         super().__init__(scripts_dir, connection)
-        if not self.database:
-            raise RuntimeError(
-                "The name of the replication database must be provided in the connection URI."
-            )
 
     def current_version(self) -> Version:
```

Comment:
it's nice that you are standardizing the pattern, `current_version` and `_set_version` can probably be moved down to the base class?

Reply:
Exactly. And not only that. I was thinking about introducing thie class:
```
class StandardReplicationMigrationManager(SchemaMigMgr):
   ...
```
Which would implement the standard way of managing migrations without reductions and complications. After that in the module-specific `schema_migration` we could have the following class factory function:
```
from lsst.qserv.schema import StandardReplicationMigrationManager

database = "qservMeta"

def make_migration_manager(
    connection: str,
    scripts_dir: str,
    set_initial_configuration: Optional[Callable[[], None]] = None
) -> SchemaMigMgr:
 
    return StandardReplicationMigrationManager(
        database,
        connection,
        scripts_dir,
        set_initial_configuration
    )
```
The next step would be to migrate the schema migration files of the relevant modules to allow using the "standard" migration protocol.

IN those special cases where we can't do we could override relevant methods of `StandardReplicationMigrationManager` as:
```
class QMetaMigrationManager(StandardReplicationMigrationManager):
    ...
```

Reply:
^ sorry, I had to fix s "bug" in the example above.

Reply:
I don't see why we need a "standard" migration manager. Just implement default behaviors in `SchemaMigMgr` and as you suggested with `StandardMigrationManager`, allow subclasses to override default behavior as needed.

Reply:
That's a very good option too. In this case, we may need to add the name of a database as a parameter of class `SchemaMigMgr`. We can't rely on the database name in the connection string because we may use the same connection string to manage different databases. A good example would be:
```
mysql://root:CHANGEME@127.0.0.1:3306?socket=/qserv/mariadb/run/mysqld.sock
```
or this one:
```
mysql://root:CHANGEME@czar-db:3306/qservMeta
```
Each could be used for managing schemas of databases:
```
qservMeta
qservCssData
qservResults
...
```
Or any other databases at czar's MariaDB.

Reply:
Upon further thinking on this improvement, I decided not to do it in this ticket. This will be postponed before we'll decided on a comprehensive solution for schema management (which I expect will also affect the schema migration tools).